src/Microsoft.TemplateEngine.Cli/NewCommandCallbacks.cs

```diff
@@ -21,5 +21,31 @@ public sealed class NewCommandCallbacks
         /// Callback to be executed to restore a project.
         /// </summary>
         public Func<string, bool>? RestoreProject { get; set; }
+
+        /// <summary>
+        /// Callback to be executed to add reference to a project.
+        /// Parameters: <br/>
+        /// - project name (string) <br/>
+        /// - arguments (IReadOnlyList&lt;string&gt;).
+        /// </summary>
+        public Func<string, IReadOnlyList<string>, bool>? AddProjectReference { get; set; }
```

Comment:
Not introduced by this PR - but should we get rid of the public setters? It is being passed all over, but we likely do not want any swaping of the initial callbacks

Reply:
this is a valid concern, I'll remove them

Reply:
changed to `init`